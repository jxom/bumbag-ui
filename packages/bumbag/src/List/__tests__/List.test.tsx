import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { List } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<List>Hello world</List>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <div>
        <List listStyleType="disc" listStylePosition="inside">
          <List.Item>Item 1</List.Item>
          <List.Item>Item 2</List.Item>
        </List>
        <List listStyleType="circle" listStylePosition="inside">
          <List.Item>Item 1</List.Item>
          <List.Item listStyleType="georgian">Item 2</List.Item>
          <List listStyleType="circle">
            <List.Item>Item 1</List.Item>
            <List.Item>Item 2</List.Item>
          </List>
        </List>
      </div>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a basic list', () => {
    const { container } = render(
      <List>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly as an ordered list', () => {
    const { container } = render(
      <List isOrdered>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly as a horizontal list', () => {
    const { container } = render(
      <List orientation="horizontal">
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<List use="div">Hello world</List>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with List props', () => {
      const { result } = renderHook(() => List.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<List>{(ListProps) => <div {...ListProps}>Hello world</div>}</List>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('overrides', () => {
  it('List.base should render correctly', () => {
    const { container } = render(
      <List overrides={{ Link: { styles: { base: { backgroundColor: 'red' } } } }}>hello world</List>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('List.base should render correctly', () => {
    const { container } = render(
      <List>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>,
      {
        theme: { List: { styles: { base: { backgroundColor: 'red' } } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('List.ordered should render correctly', () => {
    const { container } = render(
      <List isOrdered>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>,
      {
        theme: { List: { styles: { ordered: { backgroundColor: 'red' } } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('List.ordered should render correctly', () => {
    const { container } = render(
      <List orientation="horizontal">
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>,
      {
        theme: { List: { styles: { horizontal: { backgroundColor: 'red' } } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('List.Item.base should render correctly', () => {
    const { container } = render(
      <List>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>,
      {
        theme: { List: { Item: { styles: { base: { backgroundColor: 'red' } } } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for isOrdered', () => {
    const { container } = render(
      <List>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>,
      {
        theme: { List: { defaultProps: { className: 'test', isOrdered: true } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for List.Item', () => {
    const { container } = render(
      <List>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>,
      {
        theme: { List: { Item: { defaultProps: { className: 'test' } } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
