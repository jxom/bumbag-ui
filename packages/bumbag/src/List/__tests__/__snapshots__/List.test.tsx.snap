// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  list-style: unset;
  list-style-type: none;
}

.emotion-0 .emotion-0 {
  margin-left: 1rem;
}

.emotion-0 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-List"
>
  Hello world
</div>
`;

exports[`composition hook should return with List props 1`] = `
Object {
  "className": "css-4xx14y css-63it0p css-jwhr8m fp-Box fp-List",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  list-style: unset;
  list-style-type: none;
}

.emotion-0 .emotion-0 {
  margin-left: 1rem;
}

.emotion-0 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-List"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for List.Item 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-12 {
  list-style: unset;
  list-style-type: none;
}

.emotion-12 .emotion-12 {
  margin-left: 1rem;
}

.emotion-12 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

<ul
  class="emotion-12 emotion-1 emotion-2 fp-Box fp-List"
>
  <li
    class="emotion-0 emotion-1 emotion-2 test fp-Box fp-ListItem"
  >
    Item One
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 test fp-Box fp-ListItem"
  >
    Item Two
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 test fp-Box fp-ListItem"
  >
    Item Three
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 test fp-Box fp-ListItem"
  >
    Item Four
  </li>
</ul>
`;

exports[`defaultProps should render correctly for isOrdered 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-12 {
  list-style: unset;
  list-style-type: none;
}

.emotion-12 .emotion-12 {
  margin-left: 1rem;
}

.emotion-12 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-12 {
  list-style-type: decimal;
}

.emotion-12 .emotion-12 {
  list-style-type: lower-alpha;
}

.emotion-12 .emotion-12 .emotion-12 {
  list-style-type: lower-roman;
}

<ol
  class="emotion-12 emotion-1 emotion-2 test fp-Box fp-List"
>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item One
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Two
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Three
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Four
  </li>
</ol>
`;

exports[`overrides List.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  list-style: unset;
  list-style-type: none;
}

.emotion-0 .emotion-0 {
  margin-left: 1rem;
}

.emotion-0 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

<ul
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-List"
>
  hello world
</ul>
`;

exports[`props should render a basic list 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-12 {
  list-style: unset;
  list-style-type: none;
}

.emotion-12 .emotion-12 {
  margin-left: 1rem;
}

.emotion-12 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

<ul
  class="emotion-12 emotion-1 emotion-2 fp-Box fp-List"
>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item One
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Two
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Three
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Four
  </li>
</ul>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  list-style: unset;
  list-style-type: none;
}

.emotion-0 .emotion-0 {
  margin-left: 1rem;
}

.emotion-0 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

<ul
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-List"
>
  Hello world
</ul>
`;

exports[`props should render correctly as a horizontal list 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-12 {
  list-style: unset;
  list-style-type: none;
}

.emotion-12 .emotion-12 {
  margin-left: 1rem;
}

.emotion-12 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-12.emotion-12.emotion-12 li {
  display: inline-block;
  margin-bottom: unset;
}

.emotion-12.emotion-12.emotion-12 li:not(:last-child) {
  margin-right: 1rem;
}

<ul
  class="emotion-12 emotion-1 emotion-2 fp-Box fp-List"
>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item One
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Two
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Three
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Four
  </li>
</ul>
`;

exports[`props should render correctly as an ordered list 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-12 {
  list-style: unset;
  list-style-type: none;
}

.emotion-12 .emotion-12 {
  margin-left: 1rem;
}

.emotion-12 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-12 {
  list-style-type: decimal;
}

.emotion-12 .emotion-12 {
  list-style-type: lower-alpha;
}

.emotion-12 .emotion-12 .emotion-12 {
  list-style-type: lower-roman;
}

<ol
  class="emotion-12 emotion-1 emotion-2 fp-Box fp-List"
>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item One
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Two
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Three
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Four
  </li>
</ol>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-6 {
  list-style: unset;
  list-style-type: none;
}

.emotion-6 .emotion-6 {
  margin-left: 1rem;
}

.emotion-6 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-8 {
  list-style-position: inside !important;
  list-style-type: disc !important;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-26 {
  list-style-position: inside !important;
  list-style-type: circle !important;
}

.emotion-14 {
  list-style-type: georgian !important;
}

.emotion-23 {
  list-style-type: circle !important;
}

<div>
  <ul
    class="emotion-6 emotion-1 emotion-8 fp-Box fp-List"
  >
    <li
      class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
    >
      Item 1
    </li>
    <li
      class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
    >
      Item 2
    </li>
  </ul>
  <ul
    class="emotion-6 emotion-1 emotion-26 fp-Box fp-List"
  >
    <li
      class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
    >
      Item 1
    </li>
    <li
      class="emotion-0 emotion-1 emotion-14 fp-Box fp-ListItem"
    >
      Item 2
    </li>
    <ul
      class="emotion-6 emotion-1 emotion-23 fp-Box fp-List"
    >
      <li
        class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
      >
        Item 1
      </li>
      <li
        class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
      >
        Item 2
      </li>
    </ul>
  </ul>
</div>
`;

exports[`theming List.Item.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-12 {
  list-style: unset;
  list-style-type: none;
}

.emotion-12 .emotion-12 {
  margin-left: 1rem;
}

.emotion-12 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-0 {
  background-color: #ff0000;
}

<ul
  class="emotion-12 emotion-1 emotion-2 fp-Box fp-List"
>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item One
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Two
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Three
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Four
  </li>
</ul>
`;

exports[`theming List.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-12 {
  list-style: unset;
  list-style-type: none;
}

.emotion-12 .emotion-12 {
  margin-left: 1rem;
}

.emotion-12 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-12 {
  background-color: #ff0000;
}

<ul
  class="emotion-12 emotion-1 emotion-2 fp-Box fp-List"
>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item One
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Two
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Three
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Four
  </li>
</ul>
`;

exports[`theming List.ordered should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-12 {
  list-style: unset;
  list-style-type: none;
}

.emotion-12 .emotion-12 {
  margin-left: 1rem;
}

.emotion-12 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-12 {
  list-style-type: decimal;
}

.emotion-12 .emotion-12 {
  list-style-type: lower-alpha;
}

.emotion-12 .emotion-12 .emotion-12 {
  list-style-type: lower-roman;
}

.emotion-12 {
  background-color: #ff0000;
}

<ol
  class="emotion-12 emotion-1 emotion-2 fp-Box fp-List"
>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item One
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Two
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Three
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Four
  </li>
</ol>
`;

exports[`theming List.ordered should render correctly 2`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-12 {
  list-style: unset;
  list-style-type: none;
}

.emotion-12 .emotion-12 {
  margin-left: 1rem;
}

.emotion-12 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-12.emotion-12.emotion-12 li {
  display: inline-block;
  margin-bottom: unset;
}

.emotion-12.emotion-12.emotion-12 li:not(:last-child) {
  margin-right: 1rem;
}

.emotion-12 {
  background-color: #ff0000;
}

<ul
  class="emotion-12 emotion-1 emotion-2 fp-Box fp-List"
>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item One
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Two
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Three
  </li>
  <li
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ListItem"
  >
    Item Four
  </li>
</ul>
`;
